packages/stores/src/account/base.ts

```diff
@@ -0,0 +1,450 @@
+      (repo) =>
+        repo.chainName === chainNameOrId ||
+        repo.chainRecord.chain.chain_id === chainNameOrId
+    );
+
+    if (!walletRepo) {
+      throw new Error(`Chain ${chainNameOrId} is not provided.`);
+    }
+
+    walletRepo.activate();
+    return walletRepo;
+  }
+
+  /**
+   * Get the current wallet for the given chain id
+   * @param chainNameOrId - Chain Id
+   * @returns ChainWalletBase
+   */
+  getWallet(chainNameOrId: string) {
+    const walletRepo = this.getWalletRepo(chainNameOrId);
+    const wallet = walletRepo.current;
+    const txInProgress = this.txTypeInProgressByChain.get(chainNameOrId);
+
+    if (wallet) {
+      const walletWithAccountSet = wallet as ChainWalletBase &
+        UnionToIntersection<Injects[number]> & {
+          txTypeInProgress: string;
+          isReadyToSendTx: boolean;
+        };
+
+      const injectedAccountsForChain = this.getInjectedAccounts(chainNameOrId);
+
+      /**
+       * Merge the accounts into the wallet.
+       */
+      for (const key of Object.keys(injectedAccountsForChain as object)) {
+        if (
+          walletWithAccountSet[
+            key as keyof UnionToIntersection<Injects[number]>
+          ]
+        ) {
+          continue;
+        }
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        walletWithAccountSet[key] = injectedAccountsForChain[key];
+      }
+
+      walletWithAccountSet.txTypeInProgress = txInProgress ?? "";
+      walletWithAccountSet.isReadyToSendTx =
+        walletWithAccountSet.walletStatus === WalletStatus.Connected &&
+        Boolean(walletWithAccountSet.address);
+      walletWithAccountSet.activate();
+
+      return walletWithAccountSet;
+    }
+
+    return wallet;
+  }
+
+  getInjectedAccounts(
+    chainNameOrId: string
+  ): UnionToIntersection<Injects[number]> {
+    const previousInjectedAccounts = this.injectedAccounts.get(chainNameOrId);
+    if (previousInjectedAccounts) {
+      return previousInjectedAccounts;
+    }
+
+    const newInjectedAccounts = {} as UnionToIntersection<Injects[number]>;
+
+    for (let i = 0; i < this.accountSetCreators.length; i++) {
+      const fn = this.accountSetCreators[i] as Functionify<
+        [AccountStore<Injects>, ChainGetter, string],
+        Injects[number]
+      >;
+      const r = fn(this, this.chainGetter, chainNameOrId);
+
+      for (const key of Object.keys(r)) {
+        if (
+          newInjectedAccounts[key as keyof UnionToIntersection<Injects[number]>]
+        ) {
+          continue;
+        }
+
+        newInjectedAccounts[key as keyof UnionToIntersection<Injects[number]>] =
+          r[key];
+      }
+    }
+
+    this.injectedAccounts.set(chainNameOrId, newInjectedAccounts);
+    return newInjectedAccounts;
+  }
+
+  hasWallet(string: string): boolean {
+    const wallet = this.getWallet(string);
+    return Boolean(wallet);
+  }
+
+  async sign(
```

Comment:
Given this is an async function... have you looked for any opportunities to do an async import of singing utils here to reduce main bundle size? Haven't looked in depth and if you know it's not easily possible, ignore..

Reply:
It would take quite a bit of refactoring to do this change. I tried it before, but it was taking too long with no results.